File: tests/Generator/resource/php/oop/Human.php
<?php

declare(strict_types = 1);

class Human implements \JsonSerializable, \PSX\Record\RecordableInterface
{
    protected ?string $firstName = null;
    protected ?Human $parent = null;
    public function setFirstName(?string $firstName) : void
    {
        $this->firstName = $firstName;
    }
    public function getFirstName() : ?string
    {
        return $this->firstName;
    }
    public function setParent(?Human $parent) : void
    {
        $this->parent = $parent;
    }
    public function getParent() : ?Human
    {
        return $this->parent;
    }
    public function toRecord() : \PSX\Record\RecordInterface
    {
        /** @var \PSX\Record\Record<mixed> $record */
        $record = new \PSX\Record\Record();
        $record->put('firstName', $this->firstName);
        $record->put('parent', $this->parent);
        return $record;
    }
    public function jsonSerialize() : object
    {
        return (object) $this->toRecord()->getAll();
    }
}

File: tests/Generator/resource/java/oop/Human.java

import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonSetter;

public class Human {
    private String firstName;
    private Human parent;

    @JsonSetter("firstName")
    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    @JsonGetter("firstName")
    public String getFirstName() {
        return this.firstName;
    }

    @JsonSetter("parent")
    public void setParent(Human parent) {
        this.parent = parent;
    }

    @JsonGetter("parent")
    public Human getParent() {
        return this.parent;
    }
}

